ansible/roles/cri/tasks/containerd.yml

```diff
@@ -46,25 +46,32 @@
     state: present
     sysctl_file: /etc/sysctl.d/99-kubernetes-cri.conf
 
-- name: 'debian : add an apt signing key for docker'
-  apt_key:
-    url: https://download.docker.com/linux/ubuntu/gpg
-    state: present
-  when: ansible_os_family | lower == 'debian'
-
-- name: 'debian : add apt repository for stable version'
-  apt_repository:
-    repo: deb https://download.docker.com/linux/ubuntu {{ ansible_lsb.codename }} stable
-    state: present
-  when: ansible_os_family | lower == 'debian'
+# This is for compatibility with Ubuntu versions prior than 18.04 LTS, considering removing this
+# The docker repository also does not install an up to date version of containerd
+- name: Install containerd from Docker repository
+  when: ansible_distribution == 'Ubuntu' and ansible_distribution_major_version < '18'
+  block:
+  - name: 'ubuntu : add docker containerd repository GPG key'
+    apt_key:
+      url: https://download.docker.com/linux/ubuntu/gpg
+  - name: Add Docker containerd apt repository
+    apt_repository:
+      repo: "deb https://download.docker.com/linux/ubuntu {{ ansible_lsb.codename }} stable"
+      state: present
+  - name: 'ubuntu : install docker containerd container runtime'
+    apt:
+      name:
+        - containerd.io
+      state: present
+      update_cache: yes
 
-- name: 'debian : install containerd package'
+- name: 'ubuntu : install containerd container runtime'
   apt:
     name:
-      - containerd.io
+      - containerd
     state: present
     update_cache: yes
-  when: ansible_os_family | lower == 'debian'
+  when: ansible_distribution == 'Ubuntu' and ansible_distribution_major_version > '16'
```

Comment:
This will not trigger for debian based OS that aren't ubuntu. Should we tweak this to something like the following:

`when: (ansible_distribution == 'Ubuntu' and ansible_distribution_major_version > '16') || (ansible_os_family | lower == 'debian' and ansible_distribution != 'Ubuntu')`

Reply:
Or move L61 out of the block and leave it as it was previously and only add the cache in the <18 task

Reply:
Using the Debian apt repo for Ubuntu results in an outdated containerd version - I think to do this cleanly, I need to create separate automation specific for Ubuntu.

Reply:
@rkage do you need any help with this?

i can try and tackle it from a specific side if need be (i do ubuntu, you do debian...something to that effect) although that sounds a bit overkill. 

i am very excited to get this working because i've been wanting to move to containerd for a little bit now.

additionally, this will introduce complications when running the docker_cache component. we/i will likely need to take a look at that role again and rework it so that it will function if the CRI is docker or containerd.

Reply:
If dropping support for ubuntu <18.04 then the install path for Ubuntu and Debian should be the same. There doesn't seem to be a better apt repository for containerd than Docker's at the moment.

I'm going to finish this up this evening.

Reply:
I've tested this against debian10 and ubuntu, both install properly and `crictl` is able to query the containerd socket - I'll make this resolved.